content/docs/neon-auth/customization/custom-pages.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Custom Pages
+subtitle: How to build custom authentication pages with Neon Auth
+enableTableOfContents: true
+---
+
+Custom pages allow you to take full control over the layout and logic flow of authentication pages in your application. Instead of using the default pages provided by Neon Auth, you can build your own using our built-in components or low-level functions.
+
+By default, `StackHandler` creates all authentication pages you need, however, you can replace them with your own custom implementations for a more tailored user experience.
+
+<Steps>
+## Simple Example
+
+For example, if you want to create a custom sign-in page with a customized title on the top, you can create a file at `app/signin/page.tsx`:
+
+```tsx title="app/signin/page.tsx"
+import { SignIn } from '@stackframe/stack';
+
+export default function CustomSignInPage() {
+  return (
+    <div>
+      <h1>My Custom Sign In page</h1>
+      <SignIn />
+    </div>
+  );
+}
+```
+
+Then you can instruct the Stack app in `stack.ts` to use your custom sign in page:
+
+```tsx title="stack.ts"
+export const stackServerApp = new StackServerApp({
+  // ...
+  // add these three lines
+  urls: {
+    signIn: '/signin',
+  },
+});
+```
+
+You are now all set! If you visit the `/signin` page, you should see your custom sign in page. When users attempt to access a protected page or navigate to the default `/handler/sign-in` URL, they will automatically be redirected to your new custom sign-in page.
+
+For more examples, please refer to the [Examples](../customization/custom-pages.md).
```

Comment:
Broken link, and appears to (attempt to) link to self. The Stack Auth docs are also weird (they link to self), although I'm not seeing a `page-examples/` section ported here (which I think it would link to) but maybe I'm missing it.